trio_websocket/_impl.py

```diff
@@ -42,6 +45,10 @@
 logger = logging.getLogger('trio-websocket')
 
 
+class TrioWebsocketInternalError(Exception):
+    ...
+
+
```

Comment:
needs docs

> If multiple non-Cancelled-exceptions are encountered, then it will raise TrioWebSocketInternalError with the exceptiongroup as its __cause__. This should only be possible if the background task and the user context both raise exceptions. This would previously raise a MultiError with both Exceptions.

> other alternatives could include throwing out the exception from the background task, raising an ExceptionGroup with both errors, or trying to do something fancy with __cause__ or __context__.

It seems odd to raise an internal error where a user-context exception is implicated.  Perhaps we can discard the internal error with a warning message, and propagate the user exception.